### ISSUE TITLE
`import/default` fails to warn/error between TS and JS files

### ISSUE THREAD

**Issue by shanedg** [import-js/eslint-plugin-import#1536]
In our hybrid js/ts application, `import/default` successfully catches issues importing JS into JS and TS into TS but doesn't seem to handle importing TS into JS (or vice versa). Is this supported?

`exporter.ts`
```
const asdf = 'an exported thing';
export {
  asdf
}
```
`importer.js`
```
import something from './exporter';
console.log(something.asdf);
```
With the `import/default` rule enabled, I'd expect the above to warn/error on `import something from './exporter'`.

**Comment by ljharb** [comment]
Yes, I would assume that would work.

**Comment by shanedg** [comment]
@ljharb I think our issue is that we're using separate parsers for JS and TS.
`.eslintrc.js`:
```javascript
// snip
overrides: [
  {
    files: ['*.js', '*.jsx'],
    parser: 'babel-eslint'
  },
  {
    files: ['*.ts', '*.tsx'],
    parser: '@typescript-eslint/parser',
  }
],
// snip
```
I understand that the `import/parsers` setting can help but a quick experiment didn't quite work for me. `import/parsers` sets the parser for an entire file based on extension, yes? Seems unlikely that we can tell eslint to use a parser for only specific rules within a file but I believe that's more or less what I'd need for these rules to work correctly when importing across JS/TS (without consolidating our config to a single parser).

**Comment by NickHeiner** [comment]
I'm also seeing this issue.

**Comment by NickHeiner** [comment]
I'm also seeing this (separately from when I posted above) with two JS files:

**Importer**:
```js
import events from 'nf-tvui-framework/eventDispatcher/events';
```

**Exporter**:
```js
export const POINTER_BACK = 'pointerBack';
```

There is no default export, so I'd expect a lint flag.

**Comment by ljharb** [comment]
If I can get a failing test case, I'll be happy to take a look at this.

**Comment by JounQin** [comment]
Did you try https://github.com/import-js/eslint-import-resolver-typescript
